### PR TITLE
[infra] Introduce TRIXEngine package cmake config

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit introduces cmake package config for npu-engine (trix-engine) package.

Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Related issue: #8768
Draft: #8769

### CODE REVIEW

**Comment by hseok-oh** [comment]
#8769 confirmed that we can use `NPUEngine` package if package is installed, and return `NPUEngine_NOTFOUND` if package is not installed.

File: infra/nnfw/cmake/packages/NPUEngineConfig.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Looking for pre-installed NPU engine package
```
Comment (chunseoklee) [inline infra/nnfw/cmake/packages/NPUEngineConfig.cmake:1]: NPUEngine vs TRIXEngine ? Which one is better for this ?

Comment (hseok-oh) [reply infra/nnfw/cmake/packages/NPUEngineConfig.cmake:1]: Let's change name to TRIXEngine

**Review by chunseoklee — APPROVED** [review]

**Comment by glistening** [comment]
It looks good. But I wonder how the users can download trixengine sources or libraries. Will it be a guide ? Or a cmake script for downloading `trix-engine`? As I know, it requires a id/passwd for tizen.org. Am I right?

**Comment by hseok-oh** [comment]
>But I wonder how the users can download trixengine sources or libraries. Will it be a guide ?

Please check draft.

**Comment by hseok-oh** [comment]
>As I know, it requires a id/passwd for tizen.org. Am I right?

To see and download trix-engine source code, you are right.


We may allow trix-engine backend and triv loader on tizen gbs build, and x64 native build if trix-engine-dev library is installed.

**Review by jyoungyun — APPROVED** [review]
LGTM

**Comment by glistening** [comment]
> > But I wonder how the users can download trixengine sources or libraries. Will it be a guide ?
> 
> Please check draft.

I asked after checking draft. I read again but I still fail to find how to get trix-engine installed from draft.

**Comment by glistening** [comment]
> To see and download trix-engine source code, you are right.

I wondered how to get trix-engine library and headers.

> We may allow trix-engine backend and triv loader on tizen gbs build. 

Ah, it assumes tizen build either gbs build.

>  And on ubuntu, x64 native build if trix-engine-dev ubuntu library (not opened yet) is installed.

May I understand `trix-engine-dev` is prepared internally since it is not opened yet?

**Comment by hseok-oh** [comment]
>May I understand trix-engine-dev is prepared internally since it is not opened yet?

Right. `trix-engine-dev` == `npu-engine-dev`

**Review by glistening — APPROVED** [review]
LGTM

**Comment by glistening** [comment]
> > And on ubuntu, x64 native build if trix-engine-dev ubuntu library (not opened yet) is installed.
> 
> May I understand `trix-engine-dev` is prepared internally since it is not opened yet?

Precisely, `trix-engine` source is available publicly in review.tizen.org. This PR assumes the header and libraries are ready beforehand.

**Pull request merged**